doc/modules/feature_selection.rst

```diff
@@ -234,15 +234,34 @@ Randomized sparse models
 
 .. currentmodule:: sklearn.linear_model
 
-The limitation of L1-based sparse models is that faced with a group of
-very correlated features, they will select only one. To mitigate this
-problem, it is possible to use randomization techniques, reestimating the
-sparse model many times perturbing the design matrix or sub-sampling data
-and counting how many times a given regressor is selected.
+In terms of feature selection, there are some well-known limitations of
+L1-penalized models for regression and classification. For example, it is
+known that the Lasso will tend to select an individual variable out of a group
+of highly correlated features. Furthermore, even when the correlation between
+features is not too high, the conditions under which L1-penalized methods
+consistently select "good" features can be restrictive in general.
+
+To mitigate this problem, it is possible to use randomization techniques such
+as those presented in [B2009]_ and [M2010]_. The latter technique, known as
+stability selection, is implemented in the module :mod:`sklearn.linear_model`.
+In the stability selection method, a subsample of the data is fit to a
+L1-penalized model where the penalty of a random subset of coefficients has
+been scaled. Specifically, given a subsample of the data
+:math:`(y_i, x_i), i \in I`, where :math:`I \subset \{1, 2, \ldots, n\}` is a
```

Comment:
When I said "Citation / reference needed", I meant that I needed to find a reference in the scikit-learn documentation that uses `(x_i, y_i)` instead of `(y_i, x_i)`.